.pre-commit-config.yaml

```diff
@@ -25,6 +25,9 @@ repos:
       - id: check-docstring-first
       - id: check-executables-have-shebangs
       - id: check-toml
+      - id: check-case-conflict
+      - id: check-added-large-files
```

Comment:
yes, but it will fails as we have already some 400K+ files

Reply:
From https://github.com/pre-commit/pre-commit-hooks#check-added-large-files

> Limits checked files to those indicated as staged for addition by git.

So it shouldn't fail

Reply:
well it seems that the CI won't catch it at all as the CI time is already as commit...

Reply:
I didn't understand you there.
Are you saying that CI won't fail because the check is run when committing locally?

If that is it, note that not everybody uses pre-commit and we run all hooks in our CI

Reply:
> Are you saying that CI won't fail because the check is run when committing locally?

not locally, but yes, committed already

> note that not everybody uses pre-commit and we run all hooks in our CI

that is the same story, CI doe to have any `staged for addition` unless you create it again, kind of ever last commit to staging or rather move all PR commits to staging...

Reply:
there's two modes for this, the default is non-enforcing (it checks as something is committed, but they are client hooks and can be easily skipped).  the other mode is `--enforce-all` where you set hard limits and it enforces those (you can of course use `exclude: ` to allowlist files which exceed this on a case-by-case basis)

Reply:
```
docs/source/_static/images/general/fast_2.gif (2586 KB) exceeds 350 KB.
docs/source/_static/images/mnist_imgs/pt_to_pl.jpg (1712 KB) exceeds 350 KB.
docs/source/_static/images/lightning_module/pt_to_pl.png (391 KB) exceeds 350 KB.
docs/source/_static/images/general/pl_quick_start_full_compressed.gif (7687 KB) exceeds 350 KB.
docs/source/_static/images/general/pl_overview_flat.jpg (411 KB) exceeds 350 KB.
docs/source/_static/images/general/pl_overview.gif (1084 KB) exceeds 350 KB.
```

Reply:
TBH: I'd even vote for enforcing to all and keep the whitelist. Then we can also go down to something smaller (maybe 150k)? Especially since notebooks are not in here any more

Reply:
@asottile I have tried the `exclude:` but it lows only string not list, how to list multiple files or pattern?

Reply:
https://pre-commit.com/#regular-expressions